src/modm/ui/button_group.hpp

```diff
@@ -12,69 +13,85 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 // ----------------------------------------------------------------------------
+#pragma once
 
-#ifndef MODM_BUTTON_GROUP_HPP
-#define MODM_BUTTON_GROUP_HPP
-
-#include <stdint.h>
+#include <concepts>
 
 #include <modm/architecture/utils.hpp>
 #include <modm/architecture/interface/atomic_lock.hpp>
 
+#include <modm/math/utils/bit_constants.hpp>
+
 namespace modm
 {
+
+/**
+ * Button masks.
+ *
+ * Provided for convenience only.
+ * Normally it is best to define your own meaningful names for the buttons.
+ */
+enum ButtonGroupIdentifier
+{
+	NONE = 0x00,
+	BUTTON0 = Bit0,
+	BUTTON1 = Bit1,
+	BUTTON2 = Bit2,
+	BUTTON3 = Bit3,
+	BUTTON4 = Bit4,
+	BUTTON5 = Bit5,
+	BUTTON6 = Bit6,
+	BUTTON7 = Bit7,
+	ALL = 0xFF,
+};
```

Comment:
Enum class maybe? This currently leaks into the modm namespace which is quite crowded, ie. `modm::NONE` is not great.
Better just put it into the documentation, it's probably not *that* convenient… and clearly nothing in the examples breaks, so nobody seems to use it.